presto-native-execution/presto_cpp/main/tests/PrestoExchangeSourceTest.cpp

```diff
@@ -491,3 +492,79 @@ TEST_F(PrestoExchangeSourceTest, exceedingMemoryCapacityForHttpResponse) {
   ASSERT_EQ(exchangeSource->testingFailedAttempts(), 1);
   ASSERT_EQ(leafPool->getCurrentBytes(), 0);
 }
+
+TEST_F(PrestoExchangeSourceTest, memoryAllocationAndUsageCheck) {
```

Comment:
Can we have some loop and randomness in the test for a better coverage?

Reply:
This unit test checks the number of allocations and randomness things should have been covered by all the tests in PrestoExchangeSourceTest.cpp